### PR TITLE
Add pysirius client to conda-forge

### PR DESCRIPTION
**Pull request by mfleisch**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pysirius/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (synapticarbors) [inline recipes/pysirius/meta.yaml:1]: Please remove all comments that are not specific to the package being added.

**Review by synapticarbors — CHANGES_REQUESTED** [review]

**Comment by LukasScholz** [comment]
I hereby confirm that I am willing to be listed as a Maintainer

**Comment by mfleisch** [comment]
@conda-forge/staged-recipes ready for review

**Comment by mfleisch** [comment]
@conda-forge/help-python

**Comment by mfleisch** [comment]
@synapticarbors can you please have a look if the pr is ready to be merged.

File: recipes/pysirius/run_test.bat [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ECHO "### [JAVA] Try run %JAVA_HOME%"
+%JAVA_HOME%/bin/java.exe -version
+
+ECHO "### [EXE] RUN ILP SOLVER TEST"
+%PYTHON% "%RECIPE_DIR%\test_script.py"
+if errorlevel 1 exit 1
+
+ECHO "### [EXE] CHECK ILP SOLVER TEST"
+If not exist "%cd%\test_fragtree.txt" (
+    echo Framgentation tree test [EXE] failed!
```
Comment (jaimergp) [inline recipes/pysirius/run_test.bat:23]:
```suggestion
    echo Fragmentation tree test [EXE] failed!
```

File: recipes/pysirius/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+echo "### [JAVA] Try run $JAVA_HOME"
+"$JAVA_HOME/bin/java" -version
+
+echo "### [SIRIUS API] Run Sirius test script"
+$PYTHON "$RECIPE_DIR/test_script.py"
+
+echo "### [SIRIUS] Check SIRIUS test script results"
+if [ ! -f "test_fragtree.txt" ]; then
+  echo Framgentation tree test failed!
```
Comment (jaimergp) [inline recipes/pysirius/run_test.sh:25]:
```suggestion
  echo Fragmentation tree test failed!
```

File: recipes/pysirius/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
```
Comment (jaimergp) [inline recipes/pysirius/run_test.sh:2]:
```suggestion

set -euxo pipefail
```

File: recipes/pysirius/run_test.bat [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ECHO "### Start TESTING"
+ECHO "### [JAVA] Infos about JDK locations"
+ECHO "JAVA_HOME = %JAVA_HOME%"
+ECHO "JDK_HOME = %JDK_HOME%"
+ECHO "ARCH = %ARCH%"
+ECHO "OSX_ARCH = %OSX_ARCH%"
+ECHO "RECIPE_DIR = %RECIPE_DIR%"
+
+ECHO "### [JAVA] Try run java"
+java -version
```
Comment (jaimergp) [inline recipes/pysirius/run_test.bat:12]:
```suggestion
java -version
if errorlevel 1 exit 1
```

File: recipes/pysirius/run_test.bat [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+ECHO "### [JAVA] Try run %JAVA_HOME%"
+%JAVA_HOME%/bin/java.exe -version
```
Comment (jaimergp) [inline recipes/pysirius/run_test.bat:15]:
```suggestion
%JAVA_HOME%/bin/java.exe -version
if errorlevel 1 exit 1
```

**Comment by jaimergp** [comment]
Taking a look now, but CI needs to rerun, since it's been a while. In the meantime, let me know if those suggestions are useful. Just a typo and trying to catch some errors early, but nothing critical.

**Comment by joXemMx** [comment]
@jaimergp Thanks a lot for taking a look! I have included your suggestions and got the CI to run successfully again.

File: recipes/pysirius/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+test:
+  imports:
+    - PySirius
+  requires:
+    - pip
+  commands:
+    - pip check
+    - sh run_test.sh           # [not win]
+    - call run_test.bat        # [win]
```
Comment (jaimergp) [inline recipes/pysirius/meta.yaml:40]: Having these scripts named `run_test.*` tells `conda-build` (counterintuitively) to ignore the `test.commands` section. Please rename to something else like `test_script.*` or something like that so `pip check does run.

**Review by jaimergp — COMMENTED** [review]
The errors were about the `u` in `set -euxo pipefail` (unbound variables result in error). You can still do `set -ex` if you want to at least have failures error out early, but up to you (it's just usually easier to debug when things go wrong).

That aside, the logs show that you are packaging your tests as `test/*.py`, which would result in a top-level package name `test` (e.g. `import test`), which is going to cause trouble with other packages due to clobbering and other conflicts:

```
  adding 'test/__init__.py'
  adding 'test/test_account_credentials.py'
  adding 'test/test_account_info.py'
  adding 'test/test_annotated_peak.py'
  adding 'test/test_annotated_spectrum.py'
  adding 'test/test_canopus.py'
  adding 'test/test_canopus_predictions.py'
  ...
```

Your options are:

- Do not package your tests as a top-level package, but as a subpackage `PySirius.tests` or similar
- Exclude `test` in `pyproject.toml` or `setup.py`, and package them as part of the conda tests via `test.files` in `meta.yaml`.
- Do not package them at all (delete before running `pip install` or exclude from `pyproject.toml` or `setup.py`).

**Comment by joXemMx** [comment]
@jaimergp Thanks again! I decided to not package the test files you mentioned at all and also adjusted to your other suggestions. Is there something else you notice that needs changing?

File: recipes/pysirius/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+test:
+  imports:
+    - PySirius
+  requires:
+    - pip
+  commands:
+    - pip check
+    - sh $RECIPE_DIR/test_script.sh           # [not win]
+    - call %RECIPE_DIR%/test_script.bat        # [win]
```
Comment (jaimergp) [inline recipes/pysirius/meta.yaml:40]: If your package is noarch (line 14), you don't need Windows specifics, since it will be packaged on Linux only (but the resulting output is OS-agnostic).

File: recipes/pysirius/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+about:
+  home: https://github.com/boecker-lab/sirius-client-openAPI/tree/master/client-api_python
+  summary: 'PySirius: Python library for SIRIUS MS/MS analyses software.'
+  description: |
+    Python client for the SIRIUS (local) REST API (OpenAPI) that provides the full functionality of the SIRIUS 
+    MS/MS analyses software including its web based features as background service.
+  license: Apache-2.0
+  license_file: client-api_python/LICENSE.txt
+  # The doc_url and dev_url are optional.
```
Comment (jaimergp) [inline recipes/pysirius/meta.yaml:50]:
```suggestion
```

File: recipes/pysirius/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    - sirius-ms
+
+test:
+  imports:
+    - PySirius
+  requires:
+    - pip
+  commands:
+    - pip check
+    - sh $RECIPE_DIR/test_script.sh           # [not win]
```
Comment (jaimergp) [inline recipes/pysirius/meta.yaml:39]: This script relies on Java, right? I am assuming it is installed because sirius depends on it, but it doesn't hurt to be explicit about the deps you need in your tests.

File: recipes/pysirius/meta.yaml (hunk 8 above)
Comment (joXemMx) [reply recipes/pysirius/meta.yaml:40]: The package works with some files created when running SIRIUS. Due to file-paths differing on OS, we included scripts for each OS in order to make sure it works as intended. The package itself is noarch as adjustment to OS is done within its scripts.

**Comment by joXemMx** [comment]
@jaimergp Do you reckon the package is ready for merge? I realize the feedstock will not make use of the Windows test per default, but still I would like to keep it in as this allows me to temporarily enable the test during pull requests by setting the corresponding `noarch_platforms` in `conda-forge.yml`.

File: recipes/pysirius/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    - pip
+
+  run:
+    - python >=3.8
+    - certifi >=14.05.14
+    - six >=1.10
+    - python-dateutil >=2.5.3
+    - setuptools >=21.0.0
+    - urllib3 >=1.15.1
+    - sirius-ms
```
Comment (jaimergp) [inline recipes/pysirius/meta.yaml:30]: Just double checking: you don't need to be specific about `sirius-ms` versions here, right?

Comment (joXemMx) [reply recipes/pysirius/meta.yaml]: Not really at this stage, but I added it for completeness. We probably will want to raise limits for the sirius-ms version in later releases.

File: recipes/pysirius/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,57 @@
+{% set version = "0.9" %}
```
Comment (jaimergp) [inline recipes/pysirius/meta.yaml:1]: If this is version `0.9`, why is `pip` reporting:

```
Successfully installed PySirius-1.0.0
```

🤔

Comment (joXemMx) [reply recipes/pysirius/meta.yaml:1]: @jaimergp This is caused due to swagger generating the code as a "package" with version 1.0.0. It does not matter that it does that, as the conda-forge package is still successfully build as `py-sirius-ms 0.9` (see i.e. line 437 in linux_64 "Run docker build"). When generating a new client API in the future, swagger will still call it PySirius-1.0.0, but our package may be `py-sirius-ms 1.3` at that point. 
TL;DR: we can ignore swaggers versioning and use our own

Comment (joXemMx) [reply recipes/pysirius/meta.yaml:1]: but anyways, I will adjust it shortly so it matches

Comment (jaimergp) [reply recipes/pysirius/meta.yaml:1]: Yea, this is not a blocker for the merge. You can amend it in-feedstock in the future.
